From 6f152f922d47c84538209e245ae4bf3667070c9b Mon Sep 17 00:00:00 2001
From: Gustavo Duarte <gduarte@redhat.com>
Date: Tue, 4 Nov 2014 22:07:16 -0500
Subject: [powerpc] numa: use cached value of update->cpu in update_cpu_topology

Message-id: <1415138837-28185-8-git-send-email-gduarte@redhat.com>
Patchwork-id: 99391
O-Subject: [RHEL7.1 PATCH BZ 1150097 7/8] powerpc/numa: use cached value of update->cpu in update_cpu_topology
Bugzilla: 1150097
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>

BZ 1150097
Upstream Status: https://git.kernel.org/cgit/linux/kernel/git/mpe/linux.git/commit/?h=fixes&id=49f8d8c04368d2e29cd26752715367ccafec5b1d

Author: Nishanth Aravamudan <nacc@linux.vnet.ibm.com>
Date: Fri, 17 Oct 2014 17:49:44 -0700

    powerpc/numa: use cached value of update->cpu in update_cpu_topology

    There isn't any need to keep referring to update->cpu, as we've already
    checked cpu == update->cpu at this point.

    Signed-off-by: Nishanth Aravamudan <nacc@linux.vnet.ibm.com>
    Signed-off-by: Michael Ellerman <mpe@ellerman.id.au>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/mm/numa.c b/arch/powerpc/mm/numa.c
index 9123e8f..2e30700 100644
--- a/arch/powerpc/mm/numa.c
+++ b/arch/powerpc/mm/numa.c
@@ -1518,8 +1518,8 @@ static int update_cpu_topology(void *data)
   if (cpu != update->cpu)
    continue;
 
-  unmap_cpu_from_node(update->cpu);
-  map_cpu_to_node(update->cpu, update->new_nid);
+  unmap_cpu_from_node(cpu);
+  map_cpu_to_node(cpu, update->new_nid);
   vdso_getcpu_init();
  }
 
-- 
1.7.1